core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -668,8 +672,13 @@ private[deploy] class Worker(
       finishedApps += id
       maybeCleanupApplication(id)
 
-    case WorkerDecommission(_, _) =>
+    case DecommissionWorker =>
+      decommissionSelf()
+
+    case WorkerSigPWRReceived =>
       decommissionSelf()
+      // Tell master we starts decommissioning so it stops trying to launch executor/driver on us
```

Comment:
May I suggest "Tell the master that we are starting decommissioning so it stops trying to launch executor/driver on us"